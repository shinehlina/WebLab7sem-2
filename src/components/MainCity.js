import React from "react";

class MainCity extends React.Component {
  componentDidMount() {
    console.log("ss");
    this.props.getMainCity();
  }

  render() {
    console.log(this.props);
    return (
      <div>
        {this.props.data.isFetching ? <p>Загрузка...</p> : null}
        {!this.props.data.isFetching && this.props.data.error ? (
          <p>Ошибка</p>
        ) : null}
        {this.createCity()}
        <button onClick={this.props.getMainCity}>Try again</button>
      </div>
    );
  }

  createCity() {
    var weather = this.props.data.mainCityInfo;
    if (!this.props.data.isFetching && !this.props.data.error) {
      return (
        <div>
          <h2> {weather.name} </h2>
          <p>
            {" "}
            Coordinates:{" "}
            <i>
              [{weather.coord.lon} : {weather.coord.lat}]
            </i>
          </p>
          <p>
            {" "}
            Weather description: <i>{weather.weather.description}</i>
          </p>
          <p>
            {" "}
            Temperature: <i>{weather.main.temp}K </i>
          </p>
          <p>
            {" "}
            Humidity: <i>{weather.main.humidity}%</i>
          </p>
          <p>
            {" "}
            Wind: <i>{weather.wind.speed}m/c</i>
          </p>
        </div>
      );
    }
  }
}

export default MainCity;
