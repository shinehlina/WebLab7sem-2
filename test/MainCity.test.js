import React from "react";
 import MainCity from "../src/components/MainCity";
 import { shallow } from './enzyme';
 import { shallowToJson } from "enzyme-to-json";

 it("MainCity didn't changed", () => {
   let data = {
     isFetching: false,
     error: false,
     mainCityInfo: {
       coord: {
         lon: 30.33,
         lat: 60.03
       },
       weather: [
         {
           main: "Rain",
           description: "light rain",
           icon: "1"
         }
       ],
       main: {
         temp: 278.02,
         pressure: 1028,
         humidity: 93
       },
       wind: {
         speed: 3
       }
     }
   };
   const app = shallow(<MainCity data={data} getMainCity={() => undefined}/>);
   expect(shallowToJson(app)).toMatchSnapshot();
 });

 it("MainCity loading", () => {
   let data = {
     isFetching: true,
     error: false,
   };
   const app = shallow(<MainCity data={data} getMainCity={() => undefined} />);
   expect(shallowToJson(app)).toMatchSnapshot();
 });

 it("MainCity error", () => {
   let data = {
     isFetching: false,
     error: true,
   };
   const app = shallow(<MainCity data={data} getMainCity={() => undefined} />);
   expect(shallowToJson(app)).toMatchSnapshot();
 }); 