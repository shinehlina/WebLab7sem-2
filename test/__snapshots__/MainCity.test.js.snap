// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainCity didn't changed 1`] = `
<div>
  <Row>
    <Button
      node="button"
      onClick={[Function]}
      style={
        Object {
          "marginRight": "5px",
        }
      }
      waves="light"
    >
      Update geolocation
    </Button>
  </Row>
  <Row>
    <Col
      m={6}
      s={12}
    >
      <h2 />
      <img
        alt="icon"
        src="https://openweathermap.org/img/wn/1@2x.png"
      />
      <div
        style={
          Object {
            "float": "right",
            "fontSize": "3.9vw",
          }
        }
      >
        5
        หะก
      </div>
    </Col>
    <Col
      m={6}
      s={12}
    >
      <CityDescription
        humidity={93}
        latitude={60.03}
        longtitude={30.33}
        pressure={1028}
        temperature={278.02}
        windSpeed={3}
      />
    </Col>
  </Row>
</div>
`;

exports[`MainCity error 1`] = `
<div>
  <Row>
    <Button
      node="button"
      onClick={[Function]}
      style={
        Object {
          "marginRight": "5px",
        }
      }
      waves="light"
    >
      Update geolocation
    </Button>
  </Row>
  <p
    className="center-align"
  >
    Error
  </p>
</div>
`;

exports[`MainCity loading 1`] = `
<div>
  <Row>
    <Button
      node="button"
      onClick={[Function]}
      style={
        Object {
          "marginRight": "5px",
        }
      }
      waves="light"
    >
      Update geolocation
    </Button>
  </Row>
  <div
    className="center-align"
  >
    <Preloader
      active={true}
      color="blue"
      flashing={false}
      size="big"
    />
  </div>
</div>
`;
